src/middleware/auth.js

```diff
@@ -27,3 +27,8 @@ export const adminProtect = asyncHandler((req) => {
   if (req.headers['x-api-key'] === process.env.API_ACCESS_KEY) return;
   if (req.user.role !== 'ADMIN') throw new createError(403, 'You are not permitted to access this resource');
 });
+
+export const externalPartyProtect = asyncHandler((req) => {
+  if (req.headers['x-api-key'] === process.env.API_ACCESS_KEY) return;
+  if (req.user.role !== 'SPECTATOR') throw new createError(403, 'You are not permitted to access this resource');
```

Comment:
if this is used along with adminProtect neither of it will allow the call to go through, since each checks for a different role, can we refactor this into a function like `roleProtect` which takes in a list of accepted roles and checks accordingly ?